### DISCUSSION TITLE
新建页面无法新增路由 No routes matched location "/xxx"

### DISCUSSION THREAD

**Discussion by SilentFlute** [umijs/umi#12118] [Q&A] [answered]
复现步骤:

1. yarn create umi
2. 选simple app
3. 选yarn
4. 选npm

依赖安装完毕之后在`/src/pages`下新建`about.tsx`, 然后将`/src/pages/docs.tsx`的内容全部复制到`/src/pages/about.tsx`中, 最后在`/src/layouts/index.tsx`中新增一个路由入口`<Link to="/about">About</Link>`

运行`$ yarn dev`启动项目, 点击首页的`About`进入`/about`路由, 此时页面空白, 控制台出现警告: `No routes matched location "/about" `

这个问题很奇怪啊, 应该是umi的bug吧, 我从17年就陆陆续续用umi至今, 从没遇见过这种问题...我试了两次都依旧存在

**Comment by fz6m** [comment, marked as answer]
由于考虑到大部分用户最后路由表都比较多和复杂，所以现在 umi 4 默认创建出来的模板项目都默认采用配置式路由了，就是在 `.umirc.ts` 里手动配置 `routes` ，如果用了配置式路由，约定式就不好用了，所以你需要 `/about` 这个路由就需要去 `routes` 配置里新增这个路径才行。

如果你不需要配置式路由，就把配置文件里的 `routes` 删了即可。
